scripts/qa_crowbarsetup.sh

```diff
@@ -1956,23 +1941,69 @@ function proposal_set_value
     proposal_modify_value "$1" "$2" "$3" "$4" "="
 }
 
+# get the currently set value in the proposal
+function proposal_get_value
+{
+    local proposal="$1"
+    local proposaltype="$2"
+    local variable="$3"
+
+    local pfile=`get_proposal_filename "${proposal}" "${proposaltype}"`
+
+    safely rubyjsonparse "puts j${variable}" < $pfile
+}
+
 # wrapper for proposal_modify_value
 function proposal_increment_int
 {
     proposal_modify_value "$1" "$2" "$3" "$4" "+="
 }
 
+function setup_trusted_cert
+{
+    local certfile=$1
+    local keyfile=$2
+
+    if [[ $hacloud = 1 ]] ; then
+        cluster_node_assignment
+        local clusternodes_var=$(echo clusternodes${clusternameservices})
+        local controller_nodes=${!clusternodes_var}
+        local cn=$(get_cluster_vip_hostname $clusternameservices)
+    else
+        local controller_nodes=($(get_all_discovered_nodes | head -n 1))
+        local cn=$controller_nodes
+    fi
+
+    if [ ! -f $cn.key ] ; then
+        subj_alt="DNS:${cn},DNS:public-${cn}"
+        # openssl 1.1.1 supports setting the subjectAltName directly in the command line,
+        # but SLES 12 SP4 only has 1.0.2 so we need to read in the config like this
+        openssl req -x509 -newkey rsa:1024 -keyout $cn.key -out $cn.pem \
```

Comment:
can we do a simple CA trustchain here? this way we can actually validate that the cacert bundle (which is probably a common setup) works in the CI. should be the same work except that you need to do a small dance around signing a csr. I can provide the example code for that.